### PR TITLE
[SPIR-V] Add W/A for readnone function parameter attribute

### PR DESCRIPTION
**Pull request by MrSidims**
This patch partially reverts
https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1697/files until CPU runtime is updated.

Signed-off-by: Sidorov, Dmitry <dmitry.sidorov@intel.com>

### CODE REVIEW

File: llvm-spirv/lib/SPIRV/SPIRVWriter.cpp [hunk 1]
```diff
@@ -840,7 +840,7 @@ SPIRVFunction *LLVMToSPIRVBase::transFunctionDecl(Function *F) {
     if (Attrs.hasParamAttr(ArgNo, Attribute::ReadOnly))
       BA->addAttr(FunctionParameterAttributeNoWrite);
     if (Attrs.hasParamAttr(ArgNo, Attribute::ReadNone))
-      BA->addAttr(FunctionParameterAttributeNoReadWrite);
+      BA->addAttr(FunctionParameterAttributeNoWrite);
```
Comment (MrSidims) [inline llvm-spirv/lib/SPIRV/SPIRVWriter.cpp:843]:
```suggestion
      // TODO: intel/llvm customization
      // Need to return FunctionParameterAttributeNoReadWrite
      BA->addAttr(FunctionParameterAttributeNoWrite);
```

**Comment by MrSidims** [comment]
@asudarsa friendly ping

File: llvm-spirv/lib/SPIRV/SPIRVWriter.cpp [hunk 2]
```diff
@@ -840,7 +840,9 @@ SPIRVFunction *LLVMToSPIRVBase::transFunctionDecl(Function *F) {
     if (Attrs.hasParamAttr(ArgNo, Attribute::ReadOnly))
       BA->addAttr(FunctionParameterAttributeNoWrite);
     if (Attrs.hasParamAttr(ArgNo, Attribute::ReadNone))
-      BA->addAttr(FunctionParameterAttributeNoReadWrite);
+      // TODO: intel/llvm customization
```
Comment (asudarsa) [inline llvm-spirv/lib/SPIRV/SPIRVWriter.cpp:843]: Please add a link to the intel/llvm issue here in the comment.

Comment (asudarsa) [inline llvm-spirv/lib/SPIRV/SPIRVWriter.cpp:843]: I think that not adding an attribute at all (removing the if block) might be a better idea here. Does that cause some issues?

Comment (MrSidims) [inline llvm-spirv/lib/SPIRV/SPIRVWriter.cpp:843]:
```suggestion
      // TODO: intel/llvm customization
      // see https://github.com/intel/llvm/issues/7592
```

Comment (MrSidims) [reply llvm-spirv/lib/SPIRV/SPIRVWriter.cpp:843]: done

**Review by asudarsa — APPROVED** [review]
LGTM. Thanks

**Pull request merged**